src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Cipher.java

```diff
@@ -779,10 +814,18 @@ private int implDoFinal(byte[] out, int outOfs, int outLen)
             int k = 0;
             if (encrypt) {
                 if (paddingObj != null) {
+                    int startOff = 0;
+                    if (reqBlockUpdates) {
+                        startOff = bytesBuffered;
```

Comment:
Shouldn't the starting offset be the number of bytes in padBuffer, i.e. padBufferLen?

Reply:
Then no need for the assert(...) to check the starting offset value.

Reply:
padBufferLen and bytesBuffered look a bit confusing to me. My suspicion is that they have the same value every time we need them. I'll make the change you suggested and check that we have no regressions. If you believe the assertions are trivial, I'll remove them.

Reply:
Yes, I checked this with the debugger now. I'll make the change anyways.